lib/client.js

```diff
@@ -302,6 +302,64 @@ class Client {
     return Promise.resolve(this.__dataApiClient)
   }
 
+  /**
+   * Returns a decoded Avro Promise given the proper schema name and data
+   * @param {String} schemaName
+   * @param {Object|String} data
+   * @returns {Promise}
+   */
+  decodeData (schemaName, data) {
+    if (!schemaName || typeof schemaName !== 'string' || schemaName.trim() === '') {
+      log.debug('decodeData() schemaName parameter not defined');
+      return null;
```

Comment:
Yeah it can throw an Exception but I didn't see a generic Error class in place at this time. I did notice an AvroError handler but that is tied to Avro specific errors. As per the ECMAScript standard -- http://www.ecma-international.org/ecma-262/6.0/#sec-null-value

_**null value** - primitive value that represents the intentional absence of any object value_

Since this checks the absence, it seemed to match a null return vs undefined.